# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/d3d/VertexBuffer.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/renderer/d3d/VertexBuffer.cpp
--- a/gfx/angle/checkout/src/libANGLE/renderer/d3d/VertexBuffer.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/d3d/VertexBuffer.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -159,6 +159,7 @@
         mWritePosition = 0;
     }
 
+    mReservedSpace = size;
     return angle::Result::Continue;
 }
 
@@ -181,7 +182,6 @@
     checkedPosition += spaceRequired;
     ANGLE_CHECK_GL_ALLOC(GetImplAs<ContextD3D>(context), checkedPosition.IsValid());
 
-    ANGLE_TRY(reserveSpace(context, mReservedSpace));
     mReservedSpace = 0;
 
     ANGLE_TRY(mVertexBuffer->storeVertexAttributes(context, attrib, binding, currentValueType,
@@ -215,19 +215,14 @@
     // Protect against integer overflow
     ANGLE_CHECK_GL_ALLOC(GetImplAs<ContextD3D>(context), alignedRequiredSpace.IsValid());
 
-    mReservedSpace = alignedRequiredSpace.ValueOrDie();
+    ANGLE_TRY(reserveSpace(context, alignedRequiredSpace.ValueOrDie()));
 
     return angle::Result::Continue;
 }
 
 // StaticVertexBufferInterface Implementation
 StaticVertexBufferInterface::AttributeSignature::AttributeSignature()
-    : type(gl::VertexAttribType::InvalidEnum),
-      size(0),
-      stride(0),
-      normalized(false),
-      pureInteger(false),
-      offset(0)
+    : formatID(angle::FormatID::NONE), stride(0), offset(0)
 {}
 
 bool StaticVertexBufferInterface::AttributeSignature::matchesAttribute(
@@ -236,8 +231,7 @@
 {
     size_t attribStride = ComputeVertexAttributeStride(attrib, binding);
 
-    if (type != attrib.type || size != attrib.size || static_cast<GLuint>(stride) != attribStride ||
-        normalized != attrib.normalized || pureInteger != attrib.pureInteger)
+    if (formatID != attrib.format->id || static_cast<GLuint>(stride) != attribStride)
     {
         return false;
     }
@@ -250,10 +244,7 @@
 void StaticVertexBufferInterface::AttributeSignature::set(const gl::VertexAttribute &attrib,
                                                           const gl::VertexBinding &binding)
 {
-    type        = attrib.type;
-    size        = attrib.size;
-    normalized  = attrib.normalized;
-    pureInteger = attrib.pureInteger;
+    formatID = attrib.format->id;
     offset = stride = static_cast<GLuint>(ComputeVertexAttributeStride(attrib, binding));
     offset          = static_cast<size_t>(ComputeVertexAttributeOffset(attrib, binding)) %
              ComputeVertexAttributeStride(attrib, binding);